mapper/orm/src/main/java/org/hibernate/search/mapper/orm/impl/FullTextQueryImpl.java

```diff
@@ -69,11 +72,26 @@ else if ( type.equals( SearchQuery.class ) ) {
 		}
 	}
 
+	@Override
+	public TypedQuery<R> toJpaQuery() {
```

Comment:
It mainly provides more resilience to API changes in ORM, which can be useful since the `Query` contract is not meant to be implemented by third parties, thus they could add methods unexpectedly.

Also, it allows to more easily implement methods such as `getSingleResult()` in a way that throws the correct exceptions.

Since we're splitting the interfaces, I could create a wrapper class, separate from this one, and instantiate it when `toJpaQuery` is called? That would clean up the code of `SearcheQueryImpl` quite a lot.
The main drawback is that we wouldn't notice when our own interfaces deviate from JPA/ORM interfaces in an incompatible way (same signature with different return type). But I suppose it's not such a big deal, since we're unlikely to go back to making our interfaces extend JPA/ORM interfaces.

Reply:
I split the implementations and moved the adaptor to a separate class.